### PR TITLE
Remove some unused client capabilities and sync code

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
Remove some legacy stuff that clutters the API.

**Changes**
- Remove unused stuff from client capabilities
- Remove left over sync code from various places

**Issues**

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
(OpenAPI diff fails to generate because I touched BaseItemDto)

File: MediaBrowser.Model/Dto/BaseItemDto.cs [hunk 1]
```diff
@@ -85,11 +85,6 @@ public class BaseItemDto : IHasProviderIds, IItemDto, IHasServerId
 
         public string PreferredMetadataCountryCode { get; set; }
 
-        /// <summary>
-        /// Gets or sets a value indicating whether [supports synchronize].
-        /// </summary>
-        public bool? SupportsSync { get; set; }
```
Comment (nielsvanvelzen) [inline MediaBrowser.Model/Dto/BaseItemDto.cs:91]: @thornbill the web client is the only repository that actually uses this property but I believe the code using it is legacy:

https://github.com/jellyfin/jellyfin-web/blob/e7c149dd93f6195f83fc7c596f98672bfbfda6d3/src/components/itemHelper.js#L157C17-L167

https://github.com/jellyfin/jellyfin-web/blob/master/src/components/itemContextMenu.js#L293-L299

**Review by crobibero — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

File: MediaBrowser.Model/Dto/BaseItemDto.cs (hunk 1 above)
Comment (thornbill) [reply MediaBrowser.Model/Dto/BaseItemDto.cs:91]: Yeah looks like it 👍

**Review by Shadowghost — APPROVED** [review]

File: MediaBrowser.Model/Dto/BaseItemDto.cs (hunk 1 above)
Comment (nielsvanvelzen) [reply MediaBrowser.Model/Dto/BaseItemDto.cs:91]: https://github.com/jellyfin/jellyfin-web/pull/5082

**Comment by janvs0** [comment]
Hi @nielsvanvelzen this breaks logging in using jellyfin-androidtv 0.16.4 for me (but still works fine on the non-tv android client). Looking at the logcat excerpt below I think the unused capabilities also need removing from jellyfin-sdk-kotlin.

```
kotlinx.serialization.MissingFieldException: Fields [SupportsContentUploading, SupportsSync] are required for type with serial name 'org.jellyfin.sdk.model.api.ClientCapabilities', but they were missing at path: $.SessionInfo.Capabilities 
```

**Comment by nielsvanvelzen** [comment]
I expected this change to break the Android TV app yes. We will start on support for 10.9 once the first beta for it is out (although removing those fields from the request could be done now).

**Pull request merged**